sstables/compaction.cc

```diff
@@ -512,9 +520,24 @@ class compaction {
         report_start(formatted_msg);
 
         _compacting = std::move(ssts);
-        return make_sstable_reader();
+
+        auto consumer = make_interposer_consumer([this, gc_consumer = std::move(gc_consumer)] (flat_mutation_reader reader) mutable
+        {
+                using compact_mutations = compact_for_compaction<compacting_sstable_writer, GCConsumer>;
+                auto cfc = make_stable_flattened_mutations_consumer<compact_mutations>(*schema(), gc_clock::now(),
+                                             max_purgeable_func(),
+                                             get_compacting_sstable_writer(),
+                                             std::move(gc_consumer));
+
+                return seastar::async([cfc = std::move(cfc), reader = std::move(reader), this] () mutable {
```

Comment:
This will create one thread per bucket for TWCS. I thought we have a cap for the number of buckets, but I looked at the code and couldn't find it. Maybe it was one of those follow-ups that was never done.

Reply:
Thanks to a patch from @raphaelsc to the very I code I though didn't exist I found this limit:
```
static constexpr uint64_t max_data_segregation_window_count = 100;
```
100 * 128KB = 12.8MB, which is not disastrous.

Reply:
Also if it does get disastrous I get to blame you.

The main question is what happens if there are more than 100 buckets. If we lose data, that's disastrous. If it's a performance issue, it's not disastrous because this will be a very rare case.
If we end up with stuff from different timestamps in the same SSTable it's not disastrous either because the next compaction will fix it.

Reply:
Once above the limit, the interposer starts merging new buckets with the one closest to them.

Reply:
That is totally fine, we'll have moved from a 100+-bucket-in-one-sstable situation to a few-buckets-in-one-sstable that the next compaction should resolve.